### PR TITLE
Week7

### PR DESCRIPTION
**Pull request by Maher-Algepah**

### CODE REVIEW

File: 04-store-api/starter/controllers/products.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+const Product = require('../models/product')
+
+const getAllProductsStatic = async (req, res) => {
+    const result = Product.find({}).sort('-name').limit(2);
+    products = await result;
+    res.status(200).json({ products });
```
Comment (brandaogabriel7) [inline 04-store-api/starter/controllers/products.js]: You named your variable `result` but tried using one called `products`. Those names should match.

```suggestion
    const products = Product.find({}).sort('-name').limit(2);
    products = await result;
    res.status(200).json({ products });
```

File: 04-store-api/starter/db/connect.js
Comment (brandaogabriel7) [inline 04-store-api/starter/db/connect.js:1]: Is there a particular reason you removed the connect options from before?

File: 04-store-api/starter/middleware/error-handler.js
Comment (brandaogabriel7) [inline 04-store-api/starter/middleware/error-handler.js:1]: It usually isn't a good idea to return exceptions directly in your response to the user for security reasons. Doing that could expose sensitive data or implementation details that the user shouldn't know about.

**Review by brandaogabriel7 — APPROVED** [review]
You did great, Maher!

Please fix the issue with the static endpoint, though. Also, be mindful of what you choose to expose in your response to the user. Exceptions usually aren't useful for the user and they might expose sensitive application details.

File: 04-store-api/starter/db/connect.js
Comment (Maher-Algepah) [reply 04-store-api/starter/db/connect.js:1]: Yes, useNewUrlParser, useUnifiedTopology, useFindAndModify, and useCreateIndex are no longer supported options. Mongoose 6 always behaves as if useNewUrlParser, useUnifiedTopology, and useCreateIndex are true, and useFindAndModify is false.

**Review by Maher-Algepah — COMMENTED** [review]
I fixed the issues

**Pull request merged**